pkg/kubelet/cm/memorymanager/policy/policy.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/kubelet/cm/memorymanager/state"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+)
+
+// Type defines the policy type
+type Type string
+
+const (
+	// TypeNone defines policy type for the none policy
+	TypeNone Type = "none"
+	// TypeSingleNUMA defines policy type for the single NUMA policy
+	TypeSingleNUMA Type = "single-numa"
+)
+
+//TODO: we should re-use the policy interface under the CPU manager
+
+// Policy implements logic for pod container to a memory assignment.
+type Policy interface {
+	Name() string
+	Start(s state.State) error
+	// Allocate call is idempotent
+	Allocate(s state.State, pod *v1.Pod, container *v1.Container) error
```

Comment:
Do you have any idea what should be changed in memory manager, if someone introduces Deallocate interfaces for Hint Provider interface?